### DISCUSSION TITLE
v7.0.0

### DISCUSSION THREAD

**Discussion by antonmedv** [deployphp/deployer#3224] [General]
<h1 align="center">
  At last! It Is Done!
  <br><br>
  🎁 👏 🥳  Deployer v7 release!  🎉 🍾 🥂
</h1>

<p align="center"><br><br><a href="https://deployer.org"><img src="https://medv.io/assets/deployer/deployer.gif" alt="Deployer Screenshot" width="530"></a><br><br></p>

<h2 align="center">A deployment tool written in PHP with support for popular frameworks out of the box.</h2>
<br>

After more than 2 years of development, with contributions from 63 people, and the kind support of [sponsors](https://github.com/sponsors/antonmedv), I'm glad to present you with a brand new version of Deployer!

The v7 branch saw 40 betas, 9 RCs, with thousands of projects already using Deployer v7 in their production deployment.

[Getting Started](https://deployer.org/docs/7.x/getting-started)

```coffee
composer require deployer/deployer
```

<hr /><em>This discussion was created from the release <a href='https://github.com/deployphp/deployer/releases/tag/v7.0.0'>v7.0.0</a>.</em>

**Comment by rvanlaak** [comment]
Congrats @antonmedv ! 👏

**Comment by Sogl** [comment]
Great news! 
v.7 installation become easier 😄 👍

**Comment by loevgaard** [comment]
Thank you 🎉 ❤️

**Comment by yaayes** [comment]
That's some good news, congratulations 🎉🎉

**Comment by nufue** [comment]
@antonmedv
I am a bit confused by composer dependencies of version 7.0.0 - it mentions no dependencies whatsoever, starting with version v7.0.0-rc.5 - see section _requires_ at https://packagist.org/packages/deployer/deployer#v7.0.0-rc.5 vs. https://packagist.org/packages/deployer/deployer#v7.0.0-rc.4, which requires PHP of version `^8.0|^7.3`, `ext-json`, `symfony/console` etc.

Is this intented or were the requirements removed by mistake?

Thanks for you work.

**Reply by loevgaard** [reply]
I might be mistaken, but I think that's the phar :) See https://github.com/deployphp/deployer/blob/master/composer.json

**Reply by antonmedv** [reply]
This is done specially to not worry about dependencies.

**Reply by nufue** [reply]
@loevgaard @antonmedv Thanks for the clarification.

Currently the package could be installed on PHP 7.2 (`php7.2 /usr/bin/composer require deployer/deployer`) as there are no version constraints, but will fail after running `php7.2 vendor/bin/dep` with error message _Your Composer dependencies require a PHP version ">= 7.3.0"..._ (this error is generated by `platform_check.php` inside the PHAR).

That does not seem ideal to me, as in my opinion the package should at least require PHP version 7.3+ and `ext-json` to prevent installing it on older PHP version or with `json` extension disabled. The remaining dependencies `symfony/*`, `psr/http-message` etc. need not be required, as they are already packaged inside the PHAR.

**Reply by antonmedv** [reply]
https://github.com/deployphp/deployer/issues/3228

**Reply by fridtjof** [reply]
I was also confused about this coming from rc4, maybe it's worth leaving a note about this in the upgrade notes?
